static/js/components/SpinnerButton.js

```diff
@@ -21,19 +42,32 @@ export default class SpinnerButton extends React.Component {
       disabled,
       ...otherProps
     } = this.props;
+    const { recentlyClicked } = this.state;
 
     if (spinning && !disabled) {
-      if (!className) {
-        className = '';
+      if (recentlyClicked) {
+        if (!className) {
+          className = '';
+        }
+        className = `${className} disabled-with-spinner`;
+        children = <Spinner singleColor/>;
       }
-      className = `${className} disabled-with-spinner`;
-      children = <Spinner singleColor />;
       disabled = true;
     }
     if (disabled) {
       onClick = undefined;
     }
 
+    if (onClick) {
+      let oldOnClick = onClick;
+      onClick = (...args) => {
+        this.setState({
+          recentlyClicked: true
+        });
+        return oldOnClick(...args);
+      };
+    }
```

Comment:
I think it would be nice to pull this out into a separate method (`this.onClick` or something)